docs/lake/design.md

```diff
@@ -0,0 +1,657 @@
+is not relied on for causal relationships since clock skew can violate
+such an assumption).
+
+Segments are referred to by zero or more commits, where the commits
+are maintained in a commit journal described below.
+
+> While a zed lake is defined in terms of a cloud object store, it may also
+> be realized on top of a file system, which provides a convenient means for
+> local, small-scale deployments or test/debug workflows.  Thus, for simple use cases,
+> the complexity of running an object-store service may be avoided.
+
+#### Segments
+
+An immutable object is created by a single writer using a globally unique name
+with an embedded KSUID.  
+New objects are written in their entirety.  No updates, appends, or modifications
+may be made once an object exists.  Given these semantics, any such object may be
+trivially cached as its name or content never changes.
+
+Since the writer the name is globally unique and the
+resulting object is immutable, there is no possible write concurrency to manage.
+
+A segment is comprised of
+* one or two data objects (for row and/or column layout),
+* a meta-data object,
+* an optional seek index, and
+* zero or more search indexes.
+
+Data objects may be either in row form (i.e., ZNG) or column form (i.e., ZST),
+or both forms may be present as a query optimizer may choose to use whatever
+representation is more efficient.
+When both row and column data objects are present, they must contains the same
+underlying Zed data.
+
+Immutable objects are named as follows:
+
+|object type|name|
+|-----------|----|
+|column data|`<id>-<segment>-data.zst`|
+|row data|`<id>-<segment>-data.zng`|
+|row seek index|`<id>-<segment>-seek.zng`|
+|search index|`<id>-<segment>-index-<rule>.zng`|
+
+`<id>` is the KSUID of the commit, whiel
+the `<segment>` portion of the name is an integer representing the positional
+index of the segment within that commit.  Segments are number from 0 to N-1,
+where N segments define the commit.
+
+The seek index maps pool key values to seek offset in the ZNG file thereby
+allowing a scan to do a partial GET of the ZNG object when scanning only
+a subset of data.
+
+> Note the ZST format will have seekable checkpoints based on the sort key that
+> are encoded into its metadata section so there is no need to have a separate
+> seek index for the columnar object.
+
+#### Search Indexes
+
+To optimize pool scans, the lake design includes the well-known pruning
+concept, where segments of data can be skipped when it can be determined
+(either at "compile time" or "run time") that a segment of data is not
+needed by a scan, e.g., because a filter predicate would otherwise filter
+all of the data in that object.
+
+In addition to standard techniques for pruning a cloud scan (e.g., summary stats
+that can determine when a predicate would be false for every record, etc),
+search indexes can be built and attached to any data chunk using the
```

Comment:
This is the first time the word "chunk" appears in the doc, and there's 5 instances of "chunk" alongside 50 instances of "segment". I _assume_ you were making a conscious attempt to transition away from the term "chunk" (which I support) but if I'm wrong about that and both concepts persist, it might help to explain the difference.

I'm not flagging every instance of "chunk" in these review comments. Just flagging this one to raise the topic.

Reply:
Yes, should be segment.  I'll do global replace.